src/event.js

```diff
@@ -254,4 +240,25 @@ PubfoodEvent.prototype.on = function(event, fn) {
   return EventEmitter.prototype.on.apply(this, arguments);
 };
 
+/**
+ * Remove all event listeners.
+ *
+ * Removes both extended EventEmitter listeners and internal
+ * {@link pubfood#PubfoodEvent.emit} immediate listeners.
+ *
+ * @see https://github.com/primus/eventemitter3
+ *
+ * @param {string} event the event type
+ * @return {PubfoodEvent} - this
+ * @extends EventEmitter
+ * @private
+ */
+PubfoodEvent.prototype.removeAllListeners = function(event) {
+  EventEmitter.prototype.removeAllListeners.call(this, event);
+
+  this.observeImmediate_ = {};
```

Comment:
after calling the super type to remove the `event`, reset the immediate observers map.